### PR TITLE
add STMP_SECURE and changed auth config

### PR DESCRIPTION
**Pull request by olivierIllogika**
Only configure auth object in smtp config if user and password are provided. If no auth object is defined, [nodemailer assumes the connection is already authenticated](https://nodemailer.com/smtp/#authentication).

Also exposed the secure config option.

### CODE REVIEW

File: apps/server/src/integrations/environment/environment.service.ts [hunk 1]
```diff
@@ -87,6 +87,24 @@ export class EnvironmentService {
...
+  getSmtpAuth(): object {
+    let auth = undefined;
+    if (this.getSmtpUsername() && this.getSmtpPassword()) {
+      auth = {
+          user: this.getSmtpUsername(),
+          pass: this.getSmtpPassword(),
+        };
+    }
+    return auth;
+  }
```
Comment (Philipinho) [inline apps/server/src/integrations/environment/environment.service.ts]: Thank you for your contribution.

Can you move the logic to the mail provider? https://github.com/docmost/docmost/blob/main/apps/server/src/integrations/mail/providers/mail.provider.ts

Comment (olivierIllogika) [reply apps/server/src/integrations/environment/environment.service.ts]: Like so?
```diff

     switch (driver) {
       case MailOption.SMTP:
+        let auth = undefined;
+        if (environmentService.getSmtpUsername() && environmentService.getSmtpPassword()) {
+          auth = {
+              user: environmentService.getSmtpUsername(),
+              pass: environmentService.getSmtpPassword(),
+            };
+        }
         return {
           driver,
           config: {
             host: environmentService.getSmtpHost(),
             port: environmentService.getSmtpPort(),
             connectionTimeout: 30 * 1000, // 30 seconds
-            auth: environmentService.getSmtpAuth(),
+            auth,
             secure: environmentService.getSmtpSecure(),
           } as SMTPTransport.Options,
         };

```

Comment (Philipinho) [reply apps/server/src/integrations/environment/environment.service.ts]: Yep. Thank you.

**Comment by Philipinho** [comment]
Thank you for your contribution.

**Pull request merged**